test/test_linalg.py

```diff
@@ -8029,38 +8029,56 @@ def fn(torchfn, *args):
         self.assertRaises(RuntimeError, lambda: torch.lstsq(torch.randn(0, 0), torch.randn(0, 0)))
         self.assertRaises(RuntimeError, lambda: torch.lstsq(torch.randn(0,), torch.randn(0, 0)))
 
+    @dtypes(*floating_types())
     @tf32_on_and_off(0.005)
-    def test_tensordot(self, device):
-        a = torch.arange(60., device=device).reshape(3, 4, 5)
-        b = torch.arange(24., device=device).reshape(4, 3, 2)
-        c = torch.tensordot(a, b, dims=([1, 0], [0, 1])).cpu()
+    def test_tensordot(self, device, dtype):
+        # case 1: dims is a Tuple[List[int], List[int]]
+        a = torch.arange(60., dtype=dtype, device=device).reshape(3, 4, 5)
+        b = torch.arange(24., dtype=dtype, device=device).reshape(4, 3, 2)
         cn = torch.from_numpy(np.tensordot(a.cpu().numpy(), b.cpu().numpy(),
                                            axes=([1, 0], [0, 1])))
+        dims = ([1, 0], [0, 1])
+        c = torch.linalg.tensordot(a, b, dims=dims).cpu()
+        self.assertEqual(c, cn)
+
+        # case 2: dims is a 2-dimensional Tensor
+        dims = torch.tensor([[1, 0], [0, 1]])
+        c = torch.linalg.tensordot(a, b, dims=dims).cpu()
         self.assertEqual(c, cn)
 
-        cout = torch.zeros((5, 2), device=device)
-        torch.tensordot(a, b, dims=([1, 0], [0, 1]), out=cout).cpu()
+        # case 3: out parameter
+        cout = torch.zeros((5, 2), dtype=dtype, device=device)
+        torch.linalg.tensordot(a, b, dims=([1, 0], [0, 1]), out=cout).cpu()
         self.assertEqual(c, cout)
 
-        a = torch.randn(2, 3, 4, 5, device=device)
-        b = torch.randn(4, 5, 6, 7, device=device)
-        c = torch.tensordot(a, b, dims=2).cpu()
+        # case 4: dims is an integer
+        a = torch.randn(2, 3, 4, 5, dtype=dtype, device=device)
+        b = torch.randn(4, 5, 6, 7, dtype=dtype, device=device)
         cn = torch.from_numpy(np.tensordot(a.cpu().numpy(), b.cpu().numpy(),
                                            axes=2))
+        c = torch.linalg.tensordot(a, b, dims=2).cpu()
+        self.assertEqual(c, cn)
 
+        # case 5: dims is a Tensor with a single element
+        dims = torch.tensor([2])
+        c = torch.linalg.tensordot(a, b, dims=dims).cpu()
+        self.assertEqual(c, cn)
+
+        # case 6: dims is a negative number
         with self.assertRaisesRegex(RuntimeError, "expects dims >= 0"):
-            torch.tensordot(a, b, dims=-1)
+            torch.linalg.tensordot(a, b, dims=-1)
 
-        self.assertEqual(c, cn)
-        c = torch.tensordot(a, b).cpu()
+
+        # case 7: dims not specified
+        c = torch.linalg.tensordot(a, b).cpu()
         cn = torch.from_numpy(np.tensordot(a.cpu().numpy(), b.cpu().numpy()))
         self.assertEqual(c, cn)
 
-        a = torch.tensordot(torch.tensor(0.), torch.tensor(0.), 0)
+        # case 8: dims=0
+        a = torch.linalg.tensordot(torch.tensor(0.), torch.tensor(0.), 0)
```

Comment:
These tensors need to use the device and dtype, too

Reply:
right, added in d62803a2a6